packages/client/src/generation/TSClient/common.ts

```diff
@@ -277,7 +277,11 @@ type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never };
  * XOR is needed to have a real mutually exclusive union type
  * https://stackoverflow.com/questions/42123407/does-typescript-support-mutually-exclusive-types
  */
-type XOR<T, U> = (T | U) extends object ? (Without<T, U> & U) | (Without<U, T> & T) : T | U;
+type XOR<T, U> =
+  T extends object ?
+  U extends object ?
+    (Without<T, U> & U) | (Without<U, T> & T)
+  : U : T
```

Comment:
The original XOR was a copy pasta from stackoverflow, what is the change about? (add a comment?)

Reply:
Yep, the original type effectively broke type distribution with `(T | U)`, which is necessary for nested `XOR`s that are unions. It would in turn break in unexpected ways, so probably an issue on the TS side (since required fields would be required but show no error when not provided) - so caused by us, after all.